lib/src/util/number.dart

```diff
@@ -112,9 +112,141 @@ double fuzzyAssertRange(double number, int min, int max, [String? name]) {
+      numeratorUnits: ['deg']);
+}
+
+/// Returns the absolute value of [number].
+SassNumber abs(SassNumber number) {
+  return SassNumber(number.value.abs());
+}
+
+/// Returns the logarithm of [number] with respect to [base].
+SassNumber log(SassNumber number, SassNumber? base) {
+  if (base != null) {
+    return SassNumber(math.log(number.value) / math.log(base.value));
+  }
+  return SassNumber(math.log(number.value));
+}
+
+/// Returns the value of [num1] raised to the power of [num2].
+SassNumber pow(SassNumber num1, SassNumber num2) {
+  num1.assertNoUnits();
+  num2.assertNoUnits();
+  return SassNumber(math.pow(num1.value, num2.value));
+}
+
+/// Returns the arctangent for [y] and [x].
+SassNumber atan2(SassNumber y, SassNumber x) {
+  return SassNumber.withUnits(
+      math.atan2(y.value, x.convertValueToMatch(y, 'x', 'y')) * 180 / math.pi,
+      numeratorUnits: ['deg']);
+}
+
+/// Returns a rounded [number] based on a selected rounding [strategy],
+/// to the nearest integer multiple of [step].
+SassNumber roundWithStep(
+    SassString strategy, SassNumber number, SassNumber step) {
+  if (step.value == 0) return SassNumber(double.nan).coerceToMatch(number);
+  if (number.value.isInfinite ||
+      step.value == 0 ||
+      number.value.isNaN ||
+      step.value.isNaN) {
+    if (!step.value.isInfinite) {
+      return number;
+    }
+    return SassNumber(double.nan).coerceToMatch(number);
+  }
+
+  if (step.value.isInfinite) {
+    if (number.value == -0.0) return number;
+    if (strategy.text == "nearest" || strategy.text == "to-zero") {
+      return number.value >= 0
+          ? SassNumber(0).coerceToMatch(number)
+          : SassNumber(-0.0).coerceToMatch(number);
+    }
+    if (strategy.text == "up") {
+      return number.value >= 0
+          ? number.value == 0
+              ? number
+              : SassNumber(double.infinity).coerceToMatch(number)
+          : SassNumber(-0.0).coerceToMatch(number);
+    }
+    if (number.value < 0 && strategy.text == "down") {
+      return number.value <= -0.0
+          ? number.value == -0.0
+              ? number
+              : SassNumber(-double.infinity).coerceToMatch(number)
+          : SassNumber(0).coerceToMatch(number);
+    }
+
+    return SassNumber(0).coerceToMatch(number);
+  }
+
+  switch (strategy.text) {
+    case 'nearest':
+      return SassNumber((number.value / step.value).round() * step.value)
+          .coerceToMatch(number);
+    case 'up':
+      return SassNumber((number.value / step.value).ceil() * step.value)
+          .coerceToMatch(number);
+    case 'down':
+      return SassNumber((number.value / step.value).floor() * step.value)
+          .coerceToMatch(number);
+    case 'to-zero':
+      return number.value < 0
+          ? SassNumber((number.value / step.value).ceil() * step.value)
+              .coerceToMatch(number)
+          : SassNumber((number.value / step.value).floor() * step.value)
+              .coerceToMatch(number);
```

Comment:
We can't just do the math on the values and then add on `number`'s units at the end, since `number` and `step` may not always have the same units. We need to actually convert `step` into `number`'s units

Reply:
Done :)